bin/arnold

```diff
@@ -0,0 +1,1009 @@
+        _vault "${action}" "${vault}"
+    done
+}
+
+# Set a password to encrypt vault files with Ansible Vault
+function vault_pw() {
+
+    local force=0
+    local skip=0
+
+    _ensure_k8s_project_is_set
+
+    OPTS=$(getopt -o "fsh" --long "force,skip,help" -n "vault_pw" -- "$@")
+    eval set -- "$OPTS"
+
+    while true; do
+        case "${1}" in
+            -f|--force)
+                force=1; shift;;
+            -s|--skip)
+                skip=1; shift;;
+            -h|--help)
+                echo -e "Usage: vault_pw [OPTIONS]
+
+OPTIONS:
+  -f, --force
+         force overwrite if the file already exists
+  -s, --skip
+         skip a project if the secrets directory does not exist
+  -h, --help
+         print this message
+" 1>&2
+                return;;
+            --)
+                shift; break;;
+            *)
+                log error "Invalid option, see usage (-h)."
+                exit 1;;
+        esac
+    done
+
+    _set_vault_password "${force}"
+}
+
+# ---- Main ----
+
+
+OPTS=$(getopt -o "a:c:e:dDgh" --long "application:,customer:,environment:,dev,debug,gpg,help" -n "arnold" -- "$@")
+eval set -- "$OPTS"
+
+# Parse options to the `arnold` command
+while true; do
+    case "${1}" in
+        -a|--application)
+            export ARNOLD_APP="${2}"
+            shift 2;;
+        -c|--customer)
+            export ARNOLD_CUSTOMER="${2}"
+            shift 2;;
+        -e|--environment)
+            export ARNOLD_ENVIRONMENT="${2}"
+            shift 2;;
+        -d|--dev)
+            DEV_MODE=1
+            shift;;
+        -D|--debug)
+            LOG_LEVEL=LOG_LEVEL_DEBUG
+            shift;;
+        -g|--gpg)
+            USE_GPG=1
+            shift;;
+        -h|--help)
+            usage 0;;
+        --)
+            shift; break;;
+        *)
+            log error "Invalid option, see usage (-h)."
+            exit 1;;
+    esac
+done
+
+# Print usage if no COMMAND has been passed
+if [[ -z "${1}" ]]; then
+    usage 1
+fi
+
+# Check dependencies
+if ! which "${CURL}" ; then
+  log error "${CURL} is not installed."
+  log info "Please refer to the project's documentation to install ${CURL}: https://curl.se"
+  exit 15
+fi
+
+echo -e "
+  O   o--o  o   o  o-o  o     o-o
+ / \\  |   | |\\  | o   o |     |  \\
+o---o O-Oo  | \\ | |   | |     |   O
+|   | |  \\  |  \\| o   o |     |  /
+o   o o   o o   o  o-o  O---o o-o
+
+Command: ${COLOR_BLUE}${*}${COLOR_RESET}
+
+Parameters:
+  application: ${COLOR_BLUE}${ARNOLD_APP}${COLOR_RESET}
+  customer   : ${COLOR_BLUE}${ARNOLD_CUSTOMER}${COLOR_RESET}
+  environment: ${COLOR_BLUE}${ARNOLD_ENVIRONMENT}${COLOR_RESET}
+  gpg enabled: ${COLOR_BLUE}${USE_GPG}${COLOR_RESET}
+  dev mode   : ${COLOR_BLUE}${DEV_MODE}${COLOR_RESET}
+  log level  : ${COLOR_BLUE}$(_get_log_level_name)${COLOR_RESET}
+  CLI release: ${COLOR_BLUE}${ARNOLD_IMAGE_TAG}${COLOR_RESET}
+  image      : ${COLOR_BLUE}${ARNOLD_IMAGE}${COLOR_RESET}
+"
+
+LATEST_RELEASE=$(_get_latest_release)
```

Comment:
What happen if github is down ? Does the script will be usable or not because of this failure ?

Reply:
Good point! I should fix that.

Reply:
Done